src/SAML2/XML/AbstractXMLElement.php

```diff
@@ -43,7 +43,8 @@ public function __toString(): string
      */
     public function serialize(): string
     {
-        return $this->toXML()->ownerDocument->saveXML();
+        $xml = $this->toXML();
+        return $xml->ownerDocument->saveXML();
```

Comment:
Psalm complaining? If that's the case, it worries me a bit, because getting as strict as we are getting with typing, psalm should be certain that the original statement wasn't going to fail...

Reply:
No, it doesn't complain.. I thought I'd mimic buildup of the __toString()-method